classes/evasys_synchronizer.php

```diff
@@ -69,23 +96,28 @@ private function init_soap_client() {
     }
 
     private function get_course_information() {
-        $soapresult = $this->soapclient->GetCourse($this->get_evasys_courseid(), 'PUBLIC', true, true);
-        if (is_soap_fault($soapresult)) {
-            // This happens e.g. if there is no corresponding course in EvaSys.
-            return null;
+        $result = [];
+        foreach ($this->get_evasys_courseid() as $courseid) {
+            $soapresult = $this->soapclient->GetCourse($courseid, 'PUBLIC', true, true);
+            if (is_soap_fault($soapresult)) {
+                // This happens e.g. if there is no corresponding course in EvaSys.
+                return null;
+            }
+            $result[$courseid] = $soapresult;
         }
-        return $soapresult;
+        return $result;
     }
 
     /**
      * Builds array with all surveys and additional information to surveys
      * @return array of surveys with additional information
      */
-    public function get_surveys() {
-        if ($this->courseinformation === null) {
+    public function get_surveys($courseid) {
+        if ($this->courseinformation[$courseid] === null) {
             return array();
         }
-        $rawsurveys = $this->courseinformation->m_oSurveyHolder->m_aSurveys->Surveys;
+
+        $rawsurveys = $this->courseinformation[$courseid]->m_oSurveyHolder->m_aSurveys->Surveys;
```

Comment:
I get the following error here:

 Notice: Undefined property: stdClass::$Surveys in /www/data/LearnWebTest/htdocs/LearnWebTest/evasys/blocks/evasys_sync/classes/evasys_synchronizer.php on line 120